src/components/Budget/BudgetForm/BudgetExpenseClassesFields/BudgetExpenseClassesFields.js

```diff
@@ -0,0 +1,113 @@
+import React, { useMemo, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { FieldArray } from 'react-final-form-arrays';
+
+import {
+  Col,
+  Label,
+  RepeatableField,
+  Row,
+  TextField,
+} from '@folio/stripes/components';
+import {
+  FieldSelectFinal,
+  FieldSelectionFinal,
+  validateRequired,
+} from '@folio/stripes-acq-components';
+
+import {
+  EXPENSE_CLASS_STATUS,
+  EXPENSE_CLASS_STATUS_OPTIONS,
+} from '../../constants';
+
+const headLabels = (
+  <Row>
+    <Col xs>
+      <Label
+        id="budgetExpenseClassesFormNameLabel"
+        required
+      >
+        <FormattedMessage id="ui-finance.budget.expenseClasses.expenseClassName" />
+      </Label>
+    </Col>
+
+    <Col xs>
+      <Label id="budgetExpenseClassesFormAccountNumberLabel">
+        <FormattedMessage id="ui-finance.budget.expenseClasses.accountNumber" />
+      </Label>
+    </Col>
+
+    <Col xs>
+      <Label id="budgetExpenseClassesFormStatusLabel">
+        <FormattedMessage id="ui-finance.budget.expenseClasses.status" />
+      </Label>
+    </Col>
+  </Row>
+);
+
+const BudgetExpenseClassesFields = ({ expenseClasses, values }) => {
+  const expenseClassesOptions = useMemo(
+    () => expenseClasses.map(({ id, name }) => ({
+      label: name,
+      value: id,
+    })),
+    [expenseClasses],
+  );
+
+  const getAccountNumberExt = useCallback(index => (expenseClasses.find(({ id }) => (
```

Comment:
it should be useMemo and return your value

Reply:
@AndreiShumski2 please check the difference between useMemo and useCallback and then to use each one
https://stackoverflow.com/a/54963730
https://medium.com/@jan.hesters/usecallback-vs-usememo-c23ad1dc60

Reply:
I need to pass an index. Just keep it as function w/o useCB